sqlx-core/src/from_row.rs

```diff
@@ -92,6 +92,36 @@ use crate::row::Row;
 /// will set the value of the field `location` to the default value of `Option<String>`,
 /// which is `None`.
 ///
+/// ## Manual implementation
+///
+/// You can also implement the [`FromRow`] trait by hand. This can be useful if you
+/// have a struct with a field that needs manuel decoding:
+///
+/// ```rust,ignore
+/// # use sqlx_core::from_row::FromRow;
+/// # use sqlx_core::error::Result;
+/// # use sqlx_core::sqlite::{SqliteRow};
+/// # use crate::sqlx_core::row::Row;
```

Comment:
Even though this is in `sqlx-core`, since it's not built anyway I would prefer if it used these items as exported from the `sqlx` crate, as `sqlx-core` is not really meant to be used directly.

Reply:
Why is it using `ignore` though, can't it just be `no_run`?

Reply:
`ignore` makes sense if it uses the `sqlx` facade as that isn't available here, unless `dev-dependencies` can be used in doc examples?

Reply:
I'll look into this soon, I'm just quite busy right now 👍

Reply:
I'm pretty sure `dev-dependencies` can be used in doctests.

Reply:
So, I've searched a bit and found https://github.com/rust-lang/cargo/issues/1474 (which is solved). So, indeed, `dev-dependencies` can be used in doctests. Using `extern crate sqlx` we can use the "original" crate which should be used by the users. However, this introduces `sqlx` as a dev-dependency of `sqlx-core` which is kind of ugly. 
I've also looked into other doctest in the core crate and considered simply to use `use sqlx_core as sqlx` and hide it with `#` but some paths change between these two crates. For example, in `sqlx-core` `FromRow` lives at `sqlx_core::from_row::FromRow` but in the `sqlx` crate, it is `sqlx::FromRow` or `sqlx::prelude::FromRow`. This would IMO confuse an user while reading the docs.
Btw, using the `extern crate` + `dev-dependencies` approach, this test can be compiled and run. Other tests which are ignored could probably be updated too (tho I haven't looked closely into them).

Reply:
It seems that `extern crate sqlx` is actually not needed anymore.